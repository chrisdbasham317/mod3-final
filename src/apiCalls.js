export const getOrders = () => {
  return fetch('http://localhost:3001/api/v1/orders')
      .then(response => response.json())
}

export const submitOrder = (name, ingredients) => {
  const options = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      name,
      ingredients
    })
  }
  return fetch('http://localhost:3001/api/v1/orders', options)
    .then(res => {
      if (!res.ok) {
        throw Error('Unable to place error')
      }
      return res.json();
    })
}