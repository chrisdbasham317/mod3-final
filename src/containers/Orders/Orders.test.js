import React from 'react';
import { shallow } from 'enzyme';
import { Orders } from './Orders'; 
import { getOrders } from '../../apiCalls';

jest.mock('../../apiCalls');

describe('Orders', () => {
  let wrapper
  const mockOrders = [
    {
      id: 1,
      name: 'Chris',
      ingredients: ['steak', 'queso']
    },
    {
      id: 2,
      name: 'Not Chris',
      ingredients: ['chicken', 'siracha']
    }
  ]

  beforeEach(() => {
    wrapper = shallow(
      <Orders
        orders={mockOrders}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})