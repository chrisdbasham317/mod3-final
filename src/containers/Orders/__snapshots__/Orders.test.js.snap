// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="1"
  >
    <h3>
      Chris
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        steak
      </li>
      <li
        key="1"
      >
        queso
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="2"
  >
    <h3>
      Not Chris
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        chicken
      </li>
      <li
        key="1"
      >
        siracha
      </li>
    </ul>
  </div>
</section>
`;
